### PR TITLE
Minor cleanup

### PR DESCRIPTION
**Pull request by Nyholm**

### CODE REVIEW

File: src/Deferred.php [hunk 1]
```diff
@@ -14,15 +14,10 @@
 final class Deferred implements Promise
 {
     private $value;
-
```
Comment (dbu) [inline src/Deferred.php]: imho it would make sense to have phpdoc for the types of all fields here.

Comment (Nyholm) [reply src/Deferred.php]: Thank you

File: src/Deferred.php [hunk 2]
```diff
@@ -13,16 +13,34 @@
  */
 final class Deferred implements Promise
 {
+    /**
+     * @var ResponseInterface|null
+     */
     private $value;
 
+    /**
+     * @var Exception|null
```
Comment (dbu) [inline src/Deferred.php]: should this be Psr\Http\Client\ClientExceptionInterface ? (i see that for now the type declaration in the method is the httplug exception interface, but psr exception would be future proof...

Comment (Nyholm) [reply src/Deferred.php]: You are correct. Thanks

**Comment by Nyholm** [comment]
Thank you

**Pull request merged**